gr crash on invalid version

### aqua info

```console
$ aqua info

```

(current git main)

### Overview

`gr` crashes on at least some invalid versions, such as `nightly`.

### How to reproduce

aqua.yaml

```yaml

```

Other related code such as local Registry

```

```

Executed command and output

```console
$ go run ./cmd/aqua gr ClementTsang/bottom
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x784b93]

goroutine 1 [running]:
github.com/hashicorp/go-version.(*Version).String(0x0)
	.../go/pkg/mod/github.com/hashicorp/go-version@v1.7.0/version.go:386 +0x33
github.com/aquaproj/aqua/v2/pkg/controller/generate-registry.(*Group).VersionConstraint(0xc000128a50)
	.../aqua/pkg/controller/generate-registry/group.go:60 +0x87
github.com/aquaproj/aqua/v2/pkg/controller/generate-registry.mergeGroups(0xc0003faf08, {0xc000536780, 0x8, 0x7ffe599c5a9e?})
	.../aqua/pkg/controller/generate-registry/group.go:80 +0x25e
github.com/aquaproj/aqua/v2/pkg/controller/generate-registry.(*Controller).generatePackage(0xc0003d3d50?, 0xc000c49050?, 0xc0003faf08, {0x7ffe599c5a9e?, 0xc00043c278?}, {0xc0005d6820?, 0xc000056060?, 0xc000056100?})
	.../aqua/pkg/controller/generate-registry/group.go:302 +0x38
github.com/aquaproj/aqua/v2/pkg/controller/generate-registry.(*Controller).getPackageInfoWithVersionOverrides(0xc000128d20, {0x10cf428, 0xc0003f2d20}, 0xc0003d3d50, {0x7ffe599c5a9e, 0x13}, 0xc0003faf08, 0x0?, 0xc00043d0c0)
	.../aqua/pkg/controller/generate-registry/version_overrides.go:149 +0x6e7
github.com/aquaproj/aqua/v2/pkg/controller/generate-registry.(*Controller).getPackageInfoMain(0xc000128d20, {0x10cf428, 0xc0003f2d20}, 0xc0003d3d50, {0x7ffe599c5a9e?, 0xc000056f88?}, 0x0, 0xc00043d0c0)
	.../aqua/pkg/controller/generate-registry/generate.go:159 +0x639
github.com/aquaproj/aqua/v2/pkg/controller/generate-registry.(*Controller).getPackageInfo(0xc0003d0488?, {0x10cf428?, 0xc0003f2d20?}, 0x41af45?, {0x7ffe599c5a9e?, 0xe4b5a0?}, 0xc00043d160, 0xc000057008?)
	.../aqua/pkg/controller/generate-registry/generate.go:113 +0x46
github.com/aquaproj/aqua/v2/pkg/controller/generate-registry.(*Controller).genRegistry(0xc000128d20, {0x10cf428, 0xc0003f2d20}, 0xc00043d160, 0x10?, 0xd78b80?, {0x7ffe599c5a9e, 0x13})
	.../aqua/pkg/controller/generate-registry/generate.go:72 +0x65
github.com/aquaproj/aqua/v2/pkg/controller/generate-registry.(*Controller).GenerateRegistry(0xc000128d20, {0x10cf428, 0xc0003f2d20}, 0xc00043d160, 0xc0003d3d50, {0xc00049b4e0?, 0xa?, 0xb0?})
	.../aqua/pkg/controller/generate-registry/generate.go:48 +0x1a5
github.com/aquaproj/aqua/v2/pkg/cli/genr.(*command).action(0xc00011a698, {0x10cf428, 0xc0003f2d20}, 0xc0003ec788)
	.../aqua/pkg/cli/genr/command.go:146 +0x245
github.com/urfave/cli/v3.(*Command).Run(0xc0003ec788, {0x10cf428, 0xc0003f2d20}, {0xc000492060, 0x2, 0x2})
	.../go/pkg/mod/github.com/urfave/cli/v3@v3.1.1/command_run.go:326 +0x2393
github.com/urfave/cli/v3.(*Command).Run(0xc0003ecf08, {0x10cf428, 0xc0003cdb60}, {0xc0000321e0, 0x3, 0x3})
	.../go/pkg/mod/github.com/urfave/cli/v3@v3.1.1/command_run.go:265 +0x1cce
github.com/aquaproj/aqua/v2/pkg/cli.Run({0x10cf5b0, 0xc00012f880}, 0xc000128c30, {0xc0000321e0, 0x3, 0x3})
	.../aqua/pkg/cli/runner.go:110 +0xf88
main.core(0xc0003d3d50, 0xc000127860)
	.../aqua/cmd/aqua/main.go:45 +0x225
main.main()
	.../aqua/cmd/aqua/main.go:31 +0x4c
exit status 2

```


### Expected behaviour

No crash, some kind of a graceful error message.

### Actual behaviour

Crash/panic.

### Note

https://github.com/ClementTsang/bottom/releases/tag/nightly

--- a/pkg/controller/generate-registry/group.go
+++ b/pkg/controller/generate-registry/group.go
@@ -44,7 +44,7 @@ func toVersionInString(releases []*Release) string {
 	return fmt.Sprintf("Version in [%s]", strings.Join(tags, ", "))
 }
 
-func (g *Group) VersionConstraint() (string, *Release) {
+func (g *Group) VersionConstraint() (string, *Release) { //nolint:cyclop
 	switch {
 	case len(g.releases) == 1:
 		v := g.releases[0].Tag

--- a/pkg/controller/generate-registry/group.go
+++ b/pkg/controller/generate-registry/group.go
@@ -36,6 +36,14 @@ func ConvertPkgToVO(pkgInfo *registry.PackageInfo) *registry.VersionOverride {
 	}
 }
 
+func toVersionInString(releases []*Release) string {
+	tags := make([]string, len(releases))
+	for i, release := range releases {
+		tags[i] = fmt.Sprintf(`"%s"`, release.Tag)
+	}
+	return fmt.Sprintf("Version in [%s]", strings.Join(tags, ", "))
+}
+
 func (g *Group) VersionConstraint() (string, *Release) {
 	switch {
 	case len(g.releases) == 1:
@@ -46,22 +54,35 @@ func (g *Group) VersionConstraint() (string, *Release) {
 		}
 		return vc, nil
 	case g.fixed:
-		tags := make([]string, len(g.releases))
-		for i, release := range g.releases {
-			tags[i] = fmt.Sprintf(`"%s"`, release.Tag)
-		}
-		vc := fmt.Sprintf("Version in [%s]", strings.Join(tags, ", "))
+		vc := toVersionInString(g.releases)
 		if !g.pkg.Info.NoAsset {
 			return vc, g.releases[0]
 		}
 		return vc, nil
 	default:
-		release := g.releases[len(g.releases)-1]
-		vc := fmt.Sprintf(`semver("<= %s")`, strings.TrimPrefix(release.Version.String(), "v"))
-		if !g.pkg.Info.NoAsset {
-			return vc, release
+		nonSemvers := make([]*Release, 0, len(g.releases))
+		for i := len(g.releases) - 1; i >= 0; i-- {
+			release := g.releases[len(g.releases)-1]
+			if release.Version == nil {
+				nonSemvers = append(nonSemvers, release)
+				continue
+			}
+			var vc string
+			if len(nonSemvers) > 0 {
+				vc = fmt.Sprintf(`semver("<= %s") or %s`, strings.TrimPrefix(release.Version.String(), "v"), toVersionInString(nonSemvers))
+			} else {
+				vc = fmt.Sprintf(`semver("<= %s")`, strings.TrimPrefix(release.Version.String(), "v"))
+			}
+			if !g.pkg.Info.NoAsset {
+				return vc, release
+			}
+			return vc, nil
 		}
-		return vc, nil
+		vc := toVersionInString(nonSemvers)
+		if g.pkg.Info.NoAsset {
+			return vc, nil
+		}
+		return vc, g.releases[0]
 	}
 }
 

--- a/pkg/controller/generate-registry/version_overrides.go
+++ b/pkg/controller/generate-registry/version_overrides.go
@@ -63,6 +63,13 @@ func listPkgsFromVersions(pkgName string, versions []string) []*aqua.Package {
 }
 
 func excludeVersion(logE *logrus.Entry, tag string, cfg *Config) bool {
+	excludedVersions := map[string]struct{}{
+		"latest":  {},
+		"nightly": {},
+	}
+	if _, ok := excludedVersions[tag]; ok {
+		return true
+	}
 	if cfg.VersionFilter != nil {
 		f, err := expr.EvaluateVersionFilter(cfg.VersionFilter, tag)
 		if err != nil {

